src/main/java/org/skife/jdbi/v2/ClasspathStatementLocator.java

```diff
@@ -111,7 +112,7 @@ public String locate(String name, StatementContext ctx)
             if (in_stream == null) {
                 // Ensure we don't store an identity map entry which has a hard reference
                 // to the key (through the value) by copying the value, avoids potential memory leak.
-                found.put(cache_key, name == cache_key ? new String(name) : cache_key);
+                found.put(cache_key, name == cache_key ? new String(name) : name);
```

Comment:
I wondered multiple times whether any of those should also be interned but I never really came to a conclusion.